screens/GameDetails.js

```diff
@@ -0,0 +1,301 @@
+  const details = route.params.item;
+  console.log("detuals", details)
+  const [loading, setLoading] = useState(false);
+  const [accepted, setAccepted] = useState([]);
+  const [declined, setDeclined] = useState([]);
+  const [thisPlayer, setThisPlayer] = useState([]);
+
+  const [gameOrganizer, setGameOrganizer] = useState([]);
+
+  //gets current user
+  async function getPlayer() {
+    setLoading(true);
+    try {
+      const response = await Auth.currentUserInfo();
+      const player = await DataStore.query(Player, (p) => p.email.eq(response.attributes.email));
+      console.log("This player returned: ", player[0]);
+      // setThisPlayer(player[0]);
+      return player[0];
+
+
+    } catch (error) {
+      console.log("Error getting player: ", error.message);
+    }
+  }
+
+  async function getInvitedPlayers() {
+    setLoading(true);
+    let playerids = details.invited_players;
+    console.log("player ids: ", playerids);
+
+    acceptedPlayers = [];
+    declinedPlayers = [];
+    for (let i = 0; i < playerids.length; i++) {
+      let playerId = playerids[i];
+      const gamePlayer = await DataStore.query(GamePlayer, (c) => c.and(c => [c.player_id.eq(playerId), c.game_id.eq(details.id)]));
+      console.log("game player returned: ", gamePlayer);
+      const player = await DataStore.query(Player, (c) => c.id.eq(playerId));
+      console.log("player returned: ", player);
+      if (gamePlayer[0].rsvp == Rsvp.ACCEPTED) {
+        acceptedPlayers.push({ id: player[0].id, name: player[0].name, status: "In" });
+      }
+      else {
+        declinedPlayers.push({ id: player[0].id, name: player[0].name, status: "Out" });
+      }
+    }
+
+    // setAccepted(acceptedPlayers);
+    // setDeclined(declinedPlayers);
+    // setLoading(false);
+    return {
+      accepted: acceptedPlayers,
+      declined: declinedPlayers
+    }
+  }
+
+  async function getGameOrganizer(player) {
+    if (player.id == details.organizer) {
+      return player.name;
+    } else {
+      const organizer = await DataStore.query(Player, details.organizer);
+      console.log("organizer returned: ", organizer);
+      return organizer.name;
+    }
+  }
+
+
+  useEffect(() => {
+    async function fetchData() {
+      await getPlayer();
+      await getGameOrganizer();
+      await getInvitedPlayers();
+    }
+  
+    fetchData();
+  }, []);
+  
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        setLoading(true);
+  
+        const playerRes = await getPlayer();
+        setThisPlayer(playerRes);
+  
+        const organizerRes = await getGameOrganizer(playerRes);
+        setGameOrganizer(organizerRes);
+  
+        const invitedPlayersRes = await getInvitedPlayers();
+        setAccepted(invitedPlayersRes.accepted);
+        setDeclined(invitedPlayersRes.declined);
+  
+        setLoading(false);
+      } catch (error) {
+        console.log("Error fetching data: ", error);
+        setLoading(false);
```

Comment:
just put a setLoading(false) after the catch block if you don't plan on adding a return; there. We set loading to false at the end of each branch (whether we have all successful API calls or we catch an error), so just put a setLoading(false) after the catch block for consiseness.